takari-lifecycle-plugin/src/main/java/io/takari/maven/plugins/DeployParticipant.java

```diff
@@ -0,0 +1,81 @@
+package io.takari.maven.plugins;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+import java.util.Map;
+
+import org.apache.maven.AbstractMavenLifecycleParticipant;
+import org.apache.maven.MavenExecutionException;
+import org.apache.maven.execution.MavenSession;
+import org.apache.maven.plugin.descriptor.PluginDescriptor;
+import org.apache.maven.project.MavenProject;
+import org.codehaus.plexus.logging.LogEnabled;
+import org.codehaus.plexus.logging.Logger;
+import org.eclipse.aether.RepositorySystem;
+import org.eclipse.aether.deployment.DeployRequest;
+import org.eclipse.aether.deployment.DeploymentException;
+
+public class DeployParticipant extends AbstractMavenLifecycleParticipant implements LogEnabled {
+
+  private static final String CONTEXT_KEY = "deploy-at-end-requests";
+
+  protected RepositorySystem repoSystem;
+  private Logger logger;
+
+  @Override
+  public void enableLogging(Logger logger) {
+    this.logger = logger;
+  }
+
+  @Override
+  public void afterSessionEnd(MavenSession session) throws MavenExecutionException {
+    List<DeployRequest> deployAtEndRequests = getDeployAtEndRequests(session);
+    boolean noerrors = session.getResult().getExceptions().isEmpty();
+
+    if (!deployAtEndRequests.isEmpty() && noerrors) {
+      logger.info("");
+      logger.info("------------------------------------------------------------------------");
+      logger.info("-- Performing deploy at end                                           --");
+      logger.info("------------------------------------------------------------------------");
+
+      synchronized (deployAtEndRequests) {
```

Comment:
can you double-check synchronization? to me it looks either redundant (the collection is synchronizedList) or wrong (writers are not synchronized on the list). 

Personally, I would make the collection final instance variable and explicitly synchronize both readers and writers on the collection. This way correct synchronization will be much easier to assess.